tests/main/basic-target-socket-activation/task.yaml

```diff
@@ -0,0 +1,74 @@
+summary: Check that mounts are mounted correctly if snapd gets activated
+
+details: |
+  When booting with basic.target for example, snap mounts are not
+  mounted.  But sockets.target is active, which means snapd.socket
+  is. If a snap command is emitted, then snapd will start. But it
+  expects mounts to be there.
+
+systems: [ubuntu-2*]
+
+prepare: |
+  cat <<\EOF >/etc/default/grub.d/99-basic.cfg
+  # Use basic.target as default target
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.unit=basic.target"
+  # spread will need to reconnnect to ssh
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=ssh.service"
+  # ... through network
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=systemd-networkd.service"
+  # .. as a non-system user
+  GRUB_CMDLINE_LINUX_DEFAULT="${GRUB_CMDLINE_LINUX_DEFAULT} systemd.wants=systemd-user-sessions.service"
+  EOF
+
+  update-grub
+
+restore: |
+  if [ "${SPREAD_REBOOT}" = 0 ]; then
+
+    rm -f /etc/default/grub.d/99-basic.cfg
+    update-grub
+
+    if grep 'systemd[.]unit=basic[.]target' /proc/cmdline; then
+      REBOOT
+    fi
+
+  fi
+
+execute: |
+  if [ "${SPREAD_REBOOT}" = 0 ]; then
+
+    REBOOT
```

Comment:
perhaps the reboot should bo in the prepare, as the reboot is in the restore

Reply:
The problem is that we have to reboot also in the execute. Since the snapd will be started for some reason. So adding the REBOOT in prepare, does not help to remove it in execute. Less reboot, faster it is.